### PR TITLE
feat(formula): more functions component, separate plugins for calculation and UI

### PR DESCRIPTION
**Pull request by Dushusir**
- [x] Basic More Functions
- [x] separate plugins for calculation and UI

### CODE REVIEW

**Review by Jocs — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -31,4 +31,4 @@ packages/ui-plugin-slides @jikkai @wzhudev
 # formula
 packages/base-formula-engine @DR-Univer
 packages/base-numfmt-engine @Gggpound
-packages/sheets-plugin-formula @DR-Univer
+packages/sheets-plugin-formula-ui @DR-Univer
```
Comment (wzhudev) [inline .github/CODEOWNERS:34]: Add yourself as owner, too @Dushusir  :)

File: packages/base-formula-engine/src/basics/function.ts [hunk 2]
```diff
@@ -1,21 +1,83 @@
 import { BooleanNumber } from '@univerjs/core';
 
+/**
+ * Function type, refer to https://support.microsoft.com/en-us/office/excel-functions-by-category-5f91f4e9-7b42-46d2-9bd1-63f26a86c0eb
```
Comment (wzhudev) [inline packages/base-formula-engine/src/basics/function.ts:4]: A good habit to put on references

File: packages/sheets-plugin-formula-ui/src/formula-ui-plugin.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            [IFormulaPromptService, { useClass: FormulaPromptService }],
+            // controllers
+            [FormulaUIController],
+            [PromptController],
+        ];
+
+        dependencies.forEach((dependency) => this._injector.add(dependency));
+    }
+
+    override onStarting(_injector: Injector): void {}
```
Comment (wzhudev) [inline packages/sheets-plugin-formula-ui/src/formula-ui-plugin.ts:39]: Remove empty lifecycle methods.

File: .github/CODEOWNERS (hunk 1 above)
Comment (wzhudev) [reply .github/CODEOWNERS:34]: Why owners of `packages/sheets-plugin-formula` are removed?

**Review by wzhudev — APPROVED** [review]
Some minor issues need to be addressed.

File: .github/CODEOWNERS (hunk 1 above)
Comment (Dushusir) [reply .github/CODEOWNERS:34]: missed

**Review by DR-Univer — APPROVED** [review]

**Pull request merged**